frontend/src/components/common/MessageContainer.tsx

```diff
@@ -56,6 +58,7 @@ const MessageContainer = ({
           {buttonText}
         </Button>
       )}
+      {buttonComponent && <>{buttonComponent}</>}
```

Comment:
nit: I suspect you probably don't need to wrap it in a fragment? Also the first check is probably unnecessary since rendering `undefined` should do nothing.
```suggestion
      {buttonComponent}
```